### PR TITLE
fix: allow postgres role to grant usage on graphql and graphql_public schemas

### PR DESCRIPTION
**Pull request by imor**
## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

It is not possible to create a custom role and allow it to grant it access to graphql API.

## What is the new behavior?

`postgres` role is given access to the `graphql` and `graphql_public` schemas with the `with grant option` clause to allow it to grant usage permissions on a custom role.

## Additional context

See this issue in pg_graphql for context: https://github.com/supabase/pg_graphql/issues/433.

### CODE REVIEW

File: migrations/schema.sql [hunk 1]
```diff
@@ -298,6 +298,10 @@ BEGIN
         alter default privileges in schema graphql grant all on tables to postgres, anon, authenticated, service_role;
         alter default privileges in schema graphql grant all on functions to postgres, anon, authenticated, service_role;
         alter default privileges in schema graphql grant all on sequences to postgres, anon, authenticated, service_role;
+
+        -- Allow postgres role to allow granting usage on graphql and graphql_public schemas to custom roles
+        grant usage on schema graphql_public to postgres with grant option;
+        grant usage on schema graphql to postgres with grant option;
```
Comment (olirice) [inline migrations/schema.sql:304]: comparing to `migrations/db/migrations/20220404205710_pg_graphql-on-by-default.sql` this looks like the substantive change, is that right?

**Review by olirice — APPROVED** [review]
This looks great to me if that was the only change required to get it working.

Did you confirm that the postgres role was able to grant everything needed to execute a graphql request to a new role after running: 
```
grant usage on schema graphql_public to postgres with grant option;
grant usage on schema graphql to postgres with grant option;
```
on e.g. a prod instance?

If so, go ahead and merge!

File: migrations/schema.sql (hunk 1 above)
Comment (imor) [reply migrations/schema.sql:304]: > Did you confirm that the postgres role was able to grant everything needed to execute a graphql request to a new role

I tested with psql:

1. `psql -U postgres`
2. `create role new_role;`
3. `grant new_role to postgres;`
4. `set role new_role;`
5. `select graphql.resolve($${ __schema { queryType { name } } } $$);` -- fails with 'permission denied for schema graphql'.
6. `reset role;`
7. `grant usage on schema graphql to new_role;`
8. `grant usage on schema graphql_public to new_role;`
9. `set role new_role;`
10. `select graphql.resolve($${ __schema { queryType { name } } } $$);` -- succeeds

> this looks like the substantive change, is that right?

Yes, correct.

File: migrations/db/migrations/20231017062225_grant_pg_graphql_permissions_for_custom_roles.sql [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      where name = 'pg_graphql'
+  );
+
+  IF graphql_exists 
+  THEN
+  create extension if not exists pg_graphql;
+  END IF;
+END $$;
+
+-- migrate:down
```
Comment (soedirgo) [inline migrations/db/migrations/20231017062225_grant_pg_graphql_permissions_for_custom_roles.sql]: Nit: missing newline

Comment (imor) [reply migrations/db/migrations/20231017062225_grant_pg_graphql_permissions_for_custom_roles.sql]: Added.

**Comment by olirice** [comment]
LGTM!

**Review by dragarcia — APPROVED** [review]

**Pull request merged**